message.go

```diff
@@ -52,15 +57,10 @@ func (m *Message) Bitmap() *field.Bitmap {
 		return m.bitmap
 	}
 
-	bitmap, ok := m.fields[bitmapIdx]
-	if !ok {
-		return nil
-	}
-
-	if m.bitmap, ok = bitmap.(*field.Bitmap); !ok {
-		panic(fmt.Sprintf("bitmap field is %T not of type *field.Bitmap", m.fields[bitmapIdx]))
-	}
-
+	// We validate the presence and type of the bitmap field in
+	// spec.Validate() when we create the message so we can safely assume
+	// it exists and is of the correct type
+	m.bitmap, _ = m.fields[bitmapIdx].(*field.Bitmap)
 	m.bitmap.Reset()
 	m.fieldsMap[bitmapIdx] = struct{}{}
```

Comment:
Do you need to check if `m.bitmap` is non-nil before using it on L64?

Reply:
When we validate the spec we check that it defines bitmap, then NewMessage creates all fields defined in the spec, including bitmap. So we don't have to check it here, but it's a good catch!